spec/factories/attachment_references.rb

```diff
@@ -0,0 +1,20 @@
+FactoryGirl.define do
```

Comment:
I intentionally put the factory in this commit, because it depends on materials, and materials uses attachment reference only after this commit.  Or maybe I should use a single commit after this to implement the factories.